src/Hydra/Serializer/PartialCollectionViewNormalizer.php

```diff
@@ -71,12 +75,40 @@ public function normalize($object, $format = null, array $context = [])
             return $data;
         }
 
+        $metadata = isset($context['resource_class']) ? $this->resourceMetadataFactory->create($context['resource_class']) : null;
+        $isPaginatedWithCursor = $paginated && null !== $metadata && null !== $cursorPaginationAttribute = $metadata->getAttribute('pagination_via_cursor');
```

Comment:
what about using `getCollectionOperationAttribute()` instead of `getAttribute()` ?
isn't `collection_operation_name` key set in the context ?